x-pack/plugins/serverless_observability/public/components/side_navigation/index.tsx

```diff
@@ -25,136 +25,129 @@ const navigationTree: NavigationTreeDefinition = {
-                {
-                  title: i18n.translate('xpack.serverlessObservability.nav.ml.jobs', {
-                    defaultMessage: 'Anomaly detection',
-                  }),
-                  link: 'ml:anomalyDetection',
-                },
-                {
-                  title: i18n.translate('xpack.serverlessObservability.ml.logRateAnalysis', {
-                    defaultMessage: 'Log rate analysis',
-                  }),
-                  link: 'ml:logRateAnalysis',
-                  getIsActive: ({ pathNameSerialized, prepend }) => {
-                    return pathNameSerialized.includes(prepend('/app/ml/aiops/log_rate_analysis'));
-                  },
-                },
-                {
-                  title: i18n.translate('xpack.serverlessObservability.ml.changePointDetection', {
-                    defaultMessage: 'Change point detection',
-                  }),
-                  link: 'ml:changePointDetections',
-                  getIsActive: ({ pathNameSerialized, prepend }) => {
-                    return pathNameSerialized.includes(
-                      prepend('/app/ml/aiops/change_point_detection')
-                    );
-                  },
-                },
-                {
-                  title: i18n.translate('xpack.serverlessObservability.nav.ml.job.notifications', {
-                    defaultMessage: 'Job notifications',
-                  }),
-                  link: 'ml:notifications',
-                },
-              ],
+              title: i18n.translate('xpack.serverlessObservability.nav.ml.job.notifications', {
+                defaultMessage: 'Job notifications',
+              }),
+              link: 'ml:notifications',
             },
           ],
         },
         {
-          id: 'applications',
-          children: [
-            {
-              id: 'apm',
-              title: i18n.translate('xpack.serverlessObservability.nav.applications', {
-                defaultMessage: 'Applications',
-              }),
-              children: [
-                {
-                  link: 'apm:services',
-                  getIsActive: ({ pathNameSerialized, prepend }) => {
-                    const regex = /app\/apm\/.*service.*/;
-                    return regex.test(pathNameSerialized);
-                  },
-                },
-                {
-                  link: 'apm:traces',
-                  getIsActive: ({ pathNameSerialized, prepend }) => {
-                    return pathNameSerialized.startsWith(prepend('/app/apm/traces'));
-                  },
-                },
-                {
-                  link: 'apm:dependencies',
-                  getIsActive: ({ pathNameSerialized, prepend }) => {
-                    return pathNameSerialized.startsWith(prepend('/app/apm/dependencies'));
-                  },
-                },
-              ],
-            },
-          ],
+          id: 'groups-spacer-1',
+          isGroupTitle: true,
```

Comment:
I chatted with @cee-chen about the need for a nav item prop of `isSpacer`, which would improve this syntax aesthetically, with something like: `{ isSpacer: true }`

Reply:
Not sure why why we would need an extra DOM element. We should be able to add css to previous group to create spaces. The same way we add gutter between flex elements.

Reply:
> do you choose to be messy in your CSS, or do you choose to be messy in your DOM/JSX

I disagree here. This is not a question of being messy or not. Designing a navigation with items groups under different section should not have an unlimited number of options to space the groups. "Small", "medium" and "large" should be enough. There is only one side nav in an app at the end of the day 😊 And we have a very specific design we are aiming to build in Serverless.

Using DOM nodes to add spaces is, IMO, not a good choice: it affect semantic (a DOM element is used purely for layout purposes), it affects performance (large DOM trees negatively affect performance).

> For more context, there isn't a concept of "groups" anymore in the new component - an item is either an accordion or a link

This is something we might want to revisit. We discussed with the team yesterday, we might still want to be able to group navigation items without necessarily have an accordion. For example to render a bold header with a few links under it.
This should be possible by adding a few prop to the "group" (e..g `collapsible: false` --> this group is not collapsible, don't render an accordion, make the title bold and align the items on the left)

Reply:
To not block the work on this PR let's leave it like that. I will probably have a second pass at this. I don't think the solution teams should be adding fake "group-spacer" nodes in they navigation tree definition to create a nice balanced space between groups. The layout and spacing should be handled by our shared components.

One team might be tempted to add 2 fake nodes, another team 1, and before we know it all navigation look different...

Reply:
I think the second pass that Seb is commenting about would just add capabilities to the Shared UX side nav package in Kibana, and not require changes made in EUI, right? I think we all agree that we don't want EUI to be over-specialized.

Reply:
The `items` prop API of `EuiCollapsibleNavItem` is overly specialized currently - what I'll probably do is add a new generic `renderItem: () => {}` type/key to it (that replaces `isGroupTitle`) so that y'all can render whatever you need for groups and so forth (whether it's a title with a specific padding, or a specific spacer size with no title).

I'm also working on a new `EuiCollapsibleNavBeta.Group` component for you to use for "accordions" that are always open, that you can customize the spacing of as needed.

Also just to clarify, but none of these new APIs need to hold up this PR, I'm just trying to work in things into the base EUI components to make your lives easier!